File: compose/src/main/kotlin/org/dravensoft/arena/tokens/ArenaSafeArea.kt
package org.dravensoft.arena.tokens

import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.max

public object ArenaSafeArea {
    public fun top(inset: Dp, floor: Dp = ArenaTokens.sp0): Dp = max(floor, inset)

    public fun end(inset: Dp, floor: Dp = ArenaTokens.sp0): Dp = max(floor, inset)

    public fun bottom(inset: Dp, floor: Dp = ArenaTokens.sp3): Dp = max(floor, inset)

    public fun start(inset: Dp, floor: Dp = ArenaTokens.sp0): Dp = max(floor, inset)
}

File: swiftui/Sources/ArenaTokens/tokens/ArenaSafeArea.swift
import CoreGraphics

public enum ArenaSafeArea {
    public static func top(_ inset: CGFloat, floor: CGFloat = ArenaTokens.sp0) -> CGFloat { max(floor, inset) }

    public static func end(_ inset: CGFloat, floor: CGFloat = ArenaTokens.sp0) -> CGFloat { max(floor, inset) }

    public static func bottom(_ inset: CGFloat, floor: CGFloat = ArenaTokens.sp3) -> CGFloat { max(floor, inset) }

    public static func start(_ inset: CGFloat, floor: CGFloat = ArenaTokens.sp0) -> CGFloat { max(floor, inset) }
}
